### ISSUE TITLE
Genetic blindness due to eye damage

### ISSUE THREAD

**Issue by Lumipharon** [NTStation/NTstation13#914]
Revision: What ever version it is now, although I know this is a very old bug/feature, as I remember coming across this when we were testing the alt maps (efficiency/donut etc), and so has most likely been around for fuck knows how long.

Problem Description: When you go completely blind due to unprotected welding/being stabbed in the eyes/the new blindness poison etc, you gain the blindness genetic disability.

What did you expect to happen? Logically, being stabbed in the face wouldn't cause your genes to change, I expected a form of blindness related purely to eye damage.

What happened instead? How is what happened different from what you expected? Turns out screw drivers are advanced gene manipulators.

Why is this bad/What are the consequences? Why do you think this is an important issue? It makes literally no sense why this happens, and means to fix the damage to your eyes, you have to fix your SE (ryetalyn/clean SE) AND your eye damage (imidazoline or eye surgery)

Steps to reproduce the problem: Insert screw driver into eye socket, weld shift unprotected/shine laser into eyes/drink 20u of the new blinding poison, until you are COMPLETELY blind(you go temporarily blind a few times before going truly blind), then get eye surgery or imidazoline, and find that you're still utterly blind, until you drink your 0.1u of Ryetalyn.

Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): I think it's important to note - as you damage your eyes more, but BEFORE this retarded genetic blindness occurs, there is a non genetic blindness that occurs. HOWEVER, this blindness is only temporary (less then a minute), which occurs whe you receive more eye damage, once you're past a certain threshold, or something. after a while, this blindness is reduced to just 'heavily fucked up' vision, but this CAN be fixed by imidazoline just fine.

Basically someone needs to make this stop causing genetic blindness, and instead make the non genetic blindness non temporary, after enough damage.

**Comment by theOperand** [comment]
```
            on_mob_life(var/mob/living/M as mob)
                if(!M) M = holder.my_atom
                if(!data) data = 1
                switch(data)
                    if(1 to 10)
                        if(prob(19))
                            M.eye_stat += 1
                            M << "<span class='danger'>Your eyes ache.</span>"
                    if(11 to INFINITY)
                        if(prob(19))
                            M.eye_blurry = 5
                            M.eye_stat += 2
                            M << "<span class='danger'>Your eyes burn.</span>"
                            if (M.eye_stat >= 10)
                                M.disabilities |= NEARSIGHTED //honk
                                if (prob(M.eye_stat - 10 + 1) && !(M.sdisabilities & BLIND))
                                    M << "<span class='danger'>You go blind!</span>"
                                    M.sdisabilities |= BLIND //HONK
```

It's fairly obvious what went wrong here. (honks added for emphasis)

Not that eye damage code isn't retarded in itself, but hey.

**Comment by Lumipharon** [comment]
Where's that bit of code from?

**Comment by Tokiko1** [comment]
The code is from my poison. It's handled the same way with welders, eyestabing and the blindness virus symptom though.
